In [1]:
import scipy.stats as stats
from scipy.stats import chi2_contingency
import numpy as np
import pandas as pd

In [2]:
#given data
alpha=0.05

**State the Hypotheses**

a). Null Hypothesis (H₀): There is no significant association between the type of device purchased (Smart Thermostats vs. Smart Lights) and customer satisfaction level.

b). Alternative Hypothesis (H₁): There is a significant association between the type of device purchased and customer satisfaction level.

In [3]:
#Observed frequencies
observed = np.array([[50, 70],
                     [80, 100],
                     [60, 90], [30, 50],
                     [20, 50]])

**Performing chi - square statistics**

In [4]:
chi2_stat, p_value, dof, expected = stats.chi2_contingency(observed)

print("Chi-Square Statistic value:", chi2_stat)         #chi2_stats value
print("P-Value:", p_value)                        #this gives p value
print("Degrees of Freedom :", dof)                 #degree of freedom
print("Expected Frequency:",expected)             #expected frequency from observed frequency

Chi-Square Statistic value: 5.638227513227513
P-Value: 0.22784371130697179
Degrees of Freedom : 4
Expected Frequency: [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


**Critical value**

In [5]:
crit_value = stats.chi2.ppf(1 - alpha, dof)
crit_value

9.487729036781154

In [7]:
print("Critical Value at alpha = 0.05: is ", crit_value)

Critical Value at alpha = 0.05: is  9.487729036781154


**Conclusion**

P-value :- 0.22

Since the Chi-Square statistic (5.63) is less than the critical value (9.49), and the p-value (0.22) is greater than the significance level (0.05), we fail to reject the null hypothesis.

In [8]:
if chi2_stat > crit_value:
    print("Reject the null hypothesis: That means, There is a significant association between the type of device and customer satisfaction.")
else:
    print("Fail to reject the null hypothesis: There is no significant association between the type of device and customer satisfaction.")

Fail to reject the null hypothesis: There is no significant association between the type of device and customer satisfaction.


As we can conclude that Chi-square test Stats is less than critical value so we fail to reject null hypothersis, There is No Association between the type of device and customer satisfaction.